index.js

```diff
@@ -1448,14 +1447,179 @@ module.exports = {
     });
   },
 
+  /********************
+   * Raw transactions *
+   ********************/
+
+  parseRawTransactionResponse: function (rawTransactionResponse, packaged, address, privateKey, cost, callback) {
+    if (this.debug.broadcast) {
+      console.debug("[ethrpc] sendRawTransaction response:", rawTransactionResponse);
+    }
+    if (!rawTransactionResponse) {
+      if (!isFunction(callback)) return errors.RAW_TRANSACTION_ERROR;
+      return callback(errors.RAW_TRANSACTION_ERROR);
+    }
+    if (rawTransactionResponse.error) {
+      if (rawTransactionResponse.message.indexOf("rlp") > -1) {
+        var err = clone(errors.RLP_ENCODING_ERROR);
+        err.bubble = rawTransactionResponse;
+        err.packaged = packaged;
+        if (!isFunction(callback)) return err;
+        return callback(err);
+      } else if (rawTransactionResponse.message.indexOf("Nonce too low") > -1) {
+        if (this.debug.broadcast || this.debug.nonce) {
+          console.info("[ethrpc] Nonce too low, incrementing:", rawTransactionResponse.message, packaged, this.rawTxMaxNonce);
+        }
+        ++this.rawTxMaxNonce;
+        delete packaged.nonce;
+        return this.setNonceThenSubmitRawTransaction(packaged, address, privateKey, callback);
+      }
+      if (!isFunction(callback)) return rawTransactionResponse;
+      return callback(rawTransactionResponse);
+    }
+
+    // rawTransactionResponse is the txhash if nothing failed immediately
+    // (even if the tx is nulled, still index the hash)
+    this.rawTxs[rawTransactionResponse] = {tx: packaged, cost: abi.unfix(cost, "string")};
+
+    // nonce ok, complete sequence
+    if (!isFunction(callback)) return rawTransactionResponse;
+    callback(rawTransactionResponse);
+  },
+
+  submitRawTransaction: function (packaged, address, privateKey, callback) {
+    for (var rawTxHash in this.rawTxs) {
+      if (!this.rawTxs.hasOwnProperty(rawTxHash)) continue;
+      if (this.rawTxs[rawTxHash].tx.nonce === packaged.nonce && (!this.txs[rawTxHash] || this.txs[rawTxHash].status !== "failed")) {
+        packaged.nonce = abi.hex(this.rawTxMaxNonce + 1);
```

Comment:
Micah the way the nonce stuff works is that it keeps track of the highest nonce client-side so that if you dump a bunch of transactions thru ethrpc at the same time they won't all be assigned identical nonces and have to be retried.

Having nonce holes does cause a problem.  If you skip a nonce, geth will just sit there stupidly waiting for the "missing" nonce forever, and won't return any kind of error or other indication that you've done something wrong.  So the application will literally just silently stop functioning if you skip a nonce.  This is why the nonce stuff is over-engineered to the nth degree to make sure that if we make mistakes with nonces, they're too *low* rather than too high.  Low nonces are fine -- they trigger either a "nonce too low" error, a "duplicate transaction" error, or the transaction gets nulled out.  Either way it's something you can notice and react to.